src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -43,15 +44,15 @@ pub open spec fn transition_by_etcd(msg: Message, s: KubernetesAPIState) -> (Kub
         let req = msg.get_list_request();
         let s_prime = s;
         let result = Result::Err(APIError::ObjectNotFound);
-        let resp = form_msg(src, dst, list_resp_msg(result, req));
+        let resp = form_msg(src, dst, list_resp_msg(result, req, req_id));
         (s_prime, resp, Option::None)
     } else if msg.is_create_request() {
         let req = msg.get_create_request();
         if s.resources.dom().contains(req.obj.key) {
             // Creation fails
             let s_prime = s;
             let result = Result::Err(APIError::ObjectAlreadyExists);
-            let resp = form_msg(src, dst, create_resp_msg(result, req));
+            let resp = form_msg(src, dst, create_resp_msg(result, req, req_id));
```

Comment:
I'm not sure it's good to float req_id throughout the code like this. Is there no way for it to be assigned under the covers somehow in one spot?

Reply:
I agree. The current way of assigning ids is a bit error-prone.
The best spot to assign the id is in the network state machine I think.
But we need to make sure the resp has the same id as the request, which can only be done at the k8s API state machine.